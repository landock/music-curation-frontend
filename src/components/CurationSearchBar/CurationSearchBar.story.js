import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { linkTo } from '@storybook/addon-links'
import CurationSearchBar from '.'

storiesOf('CurationSearchBar', module).add('default', () => <CurationSearchBar />)