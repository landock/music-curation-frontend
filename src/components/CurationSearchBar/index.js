import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Form } from 'semantic-ui-react';

class CurationSearchBar extends PureComponent {
  render() {
    return (
      <Form inverted>
        <Form.Input icon="search" placeholder="Search Curations" />
      </Form>
    );
  }
}

CurationSearchBar.propTypes = {};

export default CurationSearchBar;
