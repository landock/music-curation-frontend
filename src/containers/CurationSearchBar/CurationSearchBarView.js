import React from 'react';
import { Form } from 'semantic-ui-react';

export default function CurationSearchBarView({ searchTerm }) {
  return (
    <Form id="search-form">
      <Form.Input
        id="search-term"
        value={searchTerm}
        icon="search"
        placeholder="Search Curations"
      />
    </Form>
  );
}
