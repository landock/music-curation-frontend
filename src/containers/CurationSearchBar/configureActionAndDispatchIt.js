import { types as searchCurationsTypes } from '../../modules/SearchCurations';
import { types as tracksTypes } from '../../modules/Tracks';
import { getData } from '../../middleware/api';

export default function configureActionAndDispatchIt(props, searchTerm) {
  let url = `/search/${searchTerm}`;
  let nextActionType = tracksTypes.TRACKS_FETCHED;

  if (!props.match) {
    nextActionType = searchCurationsTypes.SEARCH_CURATIONS_FETCHED;
    url = `/searchCurations/${searchTerm}`;
  }

  return getData(url, nextActionType);
}
