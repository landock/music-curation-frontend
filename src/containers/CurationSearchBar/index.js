import React from 'react';
import recycle from 'recycle';
import { Form } from 'semantic-ui-react';
import { types as tracksTypes } from '../../modules/Tracks';
import { types as middlewareTypes } from '../../middleware/api';

const CurationSearchBar = recycle({
  displayName: 'CurationSearchBar',
  initialState: { searchTerm: '' },
  dispatch(sources) {
    return [
      sources
        .select(Form)
        .addListener('onSubmit')
        .withLatestFrom(sources.state)
        .map(([e, state]) => {
          const { searchTerm } = state;
          return {
            type: middlewareTypes.API_REQUEST,
            nextActionType: tracksTypes.TRACKS_FETCHED,
            payload: {
              url: `/search/${searchTerm}`,
            },
          };
        }),
    ];
  },
  update(sources) {
    return [
      sources
        .select(Form.Input)
        .addListener('onChange')
        .reducer((state, e) => {
          state.searchTerm = e.target.value;
          return state;
        }),
    ];
  },
  view(props, state) {
    return (
      <Form>
        <Form.Input
          value={state.searchTerm}
          icon="search"
          placeholder="Search Curations"
        />
      </Form>
    );
  },
});

export default CurationSearchBar;
