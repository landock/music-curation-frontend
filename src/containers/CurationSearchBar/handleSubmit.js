import configureActionAndDispatchIt from './configureActionAndDispatchIt';
import { map, withLatestFrom } from 'rxjs/operators';

export default function handleSubmit(selectorStream, stateStream, propsStream) {
  console.log(selectorStream.pipe, '\n', stateStream, '\n', propsStream);
  let stream = selectorStream.pipe(
    withLatestFrom(stateStream),
    map(([e, state]) => state),
    withLatestFrom(propsStream),
    map(([state, props]) => {
      const { searchTerm } = state;
      return configureActionAndDispatchIt(props, searchTerm);
    })
  );
  return stream;
}
