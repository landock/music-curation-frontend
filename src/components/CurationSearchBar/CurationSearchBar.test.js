
import React from 'react'
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer'
import CurationSearchBar from '.'

it('CurationSearchBar: default', () => {
  const component = renderer.create(<CurationSearchBar />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})