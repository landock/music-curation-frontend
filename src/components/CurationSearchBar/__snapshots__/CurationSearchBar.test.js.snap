// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurationSearchBar: default 1`] = `
<form
  action={undefined}
  className="ui inverted form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="ui icon input"
      content={undefined}
    >
      <input
        disabled={undefined}
        onChange={[Function]}
        placeholder="Search Curations"
        required={undefined}
        tabIndex={undefined}
        type="text"
      />
      <i
        aria-hidden="true"
        className="search icon"
      />
    </div>
  </div>
</form>
`;
